# Generated by Django 5.2.1 on 2025-06-01 03:10

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('crispr', '0001_initial'),
    ]

    operations = [
        migrations.RemoveField(
            model_name='gene',
            name='chromosome',
        ),
        migrations.RemoveField(
            model_name='gene',
            name='end',
        ),
        migrations.RemoveField(
            model_name='gene',
            name='name',
        ),
        migrations.RemoveField(
            model_name='gene',
            name='sequence',
        ),
        migrations.RemoveField(
            model_name='gene',
            name='start',
        ),
        migrations.AddField(
            model_name='gene',
            name='bin',
            field=models.PositiveSmallIntegerField(blank=True, help_text='Indexing field to speed chromosome range queries', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='cdsEnd',
            field=models.PositiveIntegerField(blank=True, help_text='Coding region end (or start position for minus strand item)', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='cdsStart',
            field=models.PositiveIntegerField(blank=True, help_text='Coding region start (or end position for minus strand item)', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='chrom',
            field=models.CharField(blank=True, help_text='Reference sequence chromosome or scaffold', max_length=255, null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='exonCount',
            field=models.PositiveIntegerField(blank=True, help_text='Number of exons', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='exonEnds',
            field=models.TextField(blank=True, help_text='Exon end positions (or start positions for minus strand item)', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='exonFrames',
            field=models.TextField(blank=True, help_text='Reading frame of CDS region start of exons, 0,1,2 or -1 if no CDS', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='exonStarts',
            field=models.TextField(blank=True, help_text='Exon start positions (or end positions for minus strand item)', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='score',
            field=models.IntegerField(blank=True, help_text='score', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='txEnd',
            field=models.PositiveIntegerField(blank=True, help_text='Transcription end position (or start position for minus strand item)', null=True),
        ),
        migrations.AddField(
            model_name='gene',
            name='txStart',
            field=models.PositiveIntegerField(blank=True, help_text='Transcription start position (or end position for minus strand item)', null=True),
        ),
        migrations.AlterField(
            model_name='gene',
            name='strand',
            field=models.CharField(blank=True, choices=[('+', '+'), ('-', '-')], help_text='+ or - for strand', max_length=1, null=True),
        ),
    ]
